packages/plugins/generic-auth/src/index.ts

```diff
@@ -68,85 +87,125 @@ export type GenericAuthPluginOptions<UserType extends {} = {}, ContextType exten
        * resolves the user and inject to authenticated user into the `context`.
        * And checks for `@auth` directives usages to run validation automatically.
        */
-      mode: 'protect-auth-directive';
+      mode: 'protect-single';
```

Comment:
```suggestion
      mode: 'protect-single';
```

@saihaj @dotansimha any better alternatives for `protect-single`?

`protect-all` -> all are protected by default

`protect-single` -> more like granular configuration... `protect-manual`? `protect-flexible`?

Reply:
I think `protect-flexible` makes more sense

Reply:
I decided upon protect-granular :D